packages/errors/src/context.ts

```diff
@@ -120,8 +136,10 @@ import {
     SOLANA_ERROR__TRANSACTION_ERROR__UNKNOWN,
     SolanaErrorCode,
 } from './codes';
+import { RpcSimulateTransactionResult } from './json-rpc-error';
 
 type BasicInstructionErrorContext<T extends SolanaErrorCode> = Readonly<{ [P in T]: { index: number } }>;
+type BasicJsonRpcErrorContext<T extends SolanaErrorCode> = Readonly<{ [P in T]: { message: string } }>;
```

Comment:
Personally, I'd rather we parsed the message to grab a better context until we're able to get that information directly from the RPC. When that's fixed on the RPC layer, we can remove that parsing without breaking changes.

Reply:
We definitely can't do that.

* **Push-safety**: if I write a parser in `@solana/web3.js@X` and then the error message changes on the server, I won't be able to take back the version `X` clients that are already in the wild. They will just break.
* **Messages are non-standard**: Code numbers are core to the RPC specification. Error messages are not, and we don't want to elevate them to the level of spec.
  * Firedancer, Agave, and Sig are all likely to produce different error messages, and should be free to do so.
  * RPC implementers should be free to localize error messages in different languages.

Treating the messages themselves as non-opaque data will make the messages impossible to localize, to change, or to improve upon.

Reply:
Gotcha! Happy to disagree and commit here. 🫡